mypyc/test-data/irbuild-i64.test

```diff
@@ -0,0 +1,1370 @@
+    r0 :: ptr
+    r1 :: native_int
+    r2 :: short_int
+    r3 :: int64
+    r4 :: bit
+L0:
+    r0 = get_element_ptr a ob_size :: PyVarObject
+    r1 = load_mem r0 :: native_int*
+    keep_alive a
+    r2 = r1 << 1
+    r3 = r2 >> 1
+    r4 = y < r3 :: signed
+    if r4 goto L1 else goto L2 :: bool
+L1:
+    return 1
+L2:
+    return 0
+
+[case testMultiplyListByI64_64bit]
+from mypy_extensions import i64
+from typing import List
+
+def f(n: i64) -> List[i64]:
+    return [n] * n
+[out]
+def f(n):
+    n :: int64
+    r0 :: list
+    r1 :: object
+    r2, r3 :: ptr
+    r4, r5 :: bit
+    r6, r7, r8 :: int
+    r9 :: list
+L0:
+    r0 = PyList_New(1)
+    r1 = box(int64, n)
+    r2 = get_element_ptr r0 ob_item :: PyListObject
+    r3 = load_mem r2 :: ptr*
+    set_mem r3, r1 :: builtins.object*
+    keep_alive r0
+    r4 = n <= 4611686018427387903 :: signed
+    if r4 goto L1 else goto L2 :: bool
+L1:
+    r5 = n >= -4611686018427387904 :: signed
+    if r5 goto L3 else goto L2 :: bool
+L2:
+    r6 = CPyTagged_FromInt64(n)
+    r7 = r6
+    goto L4
+L3:
+    r8 = n << 1
+    r7 = r8
+L4:
+    r9 = CPySequence_Multiply(r0, r7)
+    return r9
+
+[case testShortIntAndI64Op]
+from mypy_extensions import i64
+from typing import List
+
+def add_i64(a: List[i64], n: i64) -> i64:
+    return len(a) + n
+def add_i64_2(a: List[i64], n: i64) -> i64:
+    return n + len(a)
+def eq_i64(a: List[i64], n: i64) -> bool:
+    if len(a) == n:
+        return True
+    return False
+def lt_i64(a: List[i64], n: i64) -> bool:
+    if n < len(a):
+        return True
+    return False
+[out]
+def add_i64(a, n):
+    a :: list
+    n :: int64
+    r0 :: ptr
+    r1 :: native_int
+    r2 :: short_int
+    r3, r4 :: int64
+L0:
+    r0 = get_element_ptr a ob_size :: PyVarObject
+    r1 = load_mem r0 :: native_int*
+    keep_alive a
+    r2 = r1 << 1
+    r3 = r2 >> 1
+    r4 = r3 + n
+    return r4
+def add_i64_2(a, n):
+    a :: list
+    n :: int64
+    r0 :: ptr
+    r1 :: native_int
+    r2 :: short_int
+    r3, r4 :: int64
+L0:
+    r0 = get_element_ptr a ob_size :: PyVarObject
+    r1 = load_mem r0 :: native_int*
+    keep_alive a
+    r2 = r1 << 1
+    r3 = r2 >> 1
+    r4 = n + r3
+    return r4
+def eq_i64(a, n):
+    a :: list
+    n :: int64
+    r0 :: ptr
+    r1 :: native_int
+    r2 :: short_int
+    r3 :: int64
+    r4 :: bit
+L0:
+    r0 = get_element_ptr a ob_size :: PyVarObject
+    r1 = load_mem r0 :: native_int*
+    keep_alive a
+    r2 = r1 << 1
+    r3 = r2 >> 1
+    r4 = r3 == n
+    if r4 goto L1 else goto L2 :: bool
+L1:
+    return 1
+L2:
+    return 0
+def lt_i64(a, n):
+    a :: list
+    n :: int64
+    r0 :: ptr
+    r1 :: native_int
+    r2 :: short_int
+    r3 :: int64
+    r4 :: bit
+L0:
+    r0 = get_element_ptr a ob_size :: PyVarObject
+    r1 = load_mem r0 :: native_int*
+    keep_alive a
+    r2 = r1 << 1
+    r3 = r2 >> 1
+    r4 = n < r3 :: signed
+    if r4 goto L1 else goto L2 :: bool
+L1:
+    return 1
+L2:
+    return 0
```

Comment:
Couple ideas for tests:
* Fixed widths types in optional types
* Unions with different int types (for whatever we can support). I am however not sure what is reasonable here,  should we support narrowing unions by comparing value to some max/min value?

Reply:
Added an optional type test case. Unions with different int types don't make much sense, I think, so I didn't add a test case (yet).